lib/qiita/markdown/filters/final_sanitizer.rb

```diff
@@ -80,6 +88,16 @@ def node
               "itemscope",
               "itemtype",
             ],
+            "p" => [
+              "class",
+              "data-default-tab",
+              "data-embed-version",
+              "data-height",
+              "data-pen-title",
+              "data-slug-hash",
+              "data-theme-id",
+              "data-user",
```

Comment:
This array duplicates `UserInputSanitizer::CODE_PEN_ATTRIBUTES`. You should define a `Qiita::Markdown::CodePen` module which contains constants corresponding to codepen.

Reply:
And please remove all data-* attributes except data-slug-hash and data-embed-version.